src/main/java/com/fasterxml/jackson/databind/deser/std/StdValueInstantiator.java

```diff
@@ -345,37 +361,106 @@ public Object createFromInt(DeserializationContext ctxt, int value) throws IOExc
                         arg, rewrapCtorProblem(ctxt, t0));
             }
         }
+
+        if (_fromBigIntegerCreator != null) {
+            Object arg = BigInteger.valueOf(value);
+            try {
+                return _fromBigIntegerCreator.call1(arg);
+            } catch (Throwable t0) {
+                return ctxt.handleInstantiationProblem(_fromBigIntegerCreator.getDeclaringClass(),
+                                                       arg, rewrapCtorProblem(ctxt, t0)
+                );
+            }
+        }
+
         return super.createFromInt(ctxt, value);
     }
 
     @Override
     public Object createFromLong(DeserializationContext ctxt, long value) throws IOException
     {
-        if (_fromLongCreator == null) {
-            return super.createFromLong(ctxt, value);
+        if (_fromLongCreator != null) {
+            Object arg = Long.valueOf(value);
+            try {
+                return _fromLongCreator.call1(arg);
+            } catch (Throwable t0) {
+                return ctxt.handleInstantiationProblem(_fromLongCreator.getDeclaringClass(),
+                                                       arg,
+                                                       rewrapCtorProblem(ctxt, t0)
+                );
+            }
         }
-        Object arg = Long.valueOf(value);
-        try {
-            return _fromLongCreator.call1(arg);
-        } catch (Throwable t0) {
-            return ctxt.handleInstantiationProblem(_fromLongCreator.getDeclaringClass(),
-                    arg, rewrapCtorProblem(ctxt, t0));
+
+        if (_fromBigIntegerCreator != null) {
+            Object arg = BigInteger.valueOf(value);
+            try {
+                return _fromBigIntegerCreator.call1(arg);
+            } catch (Throwable t0) {
+                return ctxt.handleInstantiationProblem(_fromBigIntegerCreator.getDeclaringClass(),
+                                                       arg, rewrapCtorProblem(ctxt, t0)
+                );
+            }
         }
+
+        return super.createFromLong(ctxt, value);
+    }
+
+    @Override
+    public Object createFromBigInteger(DeserializationContext ctxt, BigInteger value) throws IOException
+    {
+        if (_fromBigDecimalCreator != null) {
```

Comment:
... and here we have a bug. 

Which is not caught by test cases since numbers used do not extend beyond value range of `long` (for `BigInteger`), or, for `BigDecimal` as that is never really reported by JSON parser. Would be exposed by Smile codec, maybe some other binary codecs.
But I'll try fixing BigInteger case at least.